### PR TITLE
Request: Fix http.Client race issue when setting transport layer proxy and timeouts

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Pulled over from #779 

There is a race when we set client timeout and proxy for the requester and this will make sure we can set them both correctly.

ALSO: Makes sure we do not share http client pointers across different subsystems.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Only minor changes from me. Nice work with the amount of protections here. My earlier issue in the fee PR has been addressed as a bad Requester does not allow an exchange to be enabled 🎉

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes!

File: exchanges/request/client_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+)
+
+// this doesn't need to be included in binary
+func (c *clientTracker) contains(ohhhWOOOOOOOW *http.Client) bool {
```
Comment (thrasher-) [inline exchanges/request/client_test.go:14]: 👀

Comment (shazbert) [reply exchanges/request/client_test.go]: 🦅 fixed

**Pull request merged**